charts/tekton-pipelines/templates/020-pipelines.yaml

```diff
@@ -11,9 +11,15 @@ metadata:
     heritage: {{ $.Release.Service }}
 spec:
 
+  {{- if .workspaces -}}
+  {{ if gt (len .workspaces) 0 }}
   workspaces:
-    - name: shared-workspace
-    
+    {{ range .workspaces }}
```

Comment:
Would consider using .workspaces | toYaml | indent 4 here but should work fine

Reply:
It looks like .workspaces contain some extra fields that are not used in this template, but are used in TriggerTemplate to select storage for workspaces:

    workspaces:
      - name: shared-workspace
        volumeClaimTemplate: 2Gi
      - name: input
      - name: pr-content
        volumeClaimTemplate: 50Mi

see lines 35-63 in triggerTemplate.yaml